pkgs/development/python-modules/playwright/default.nix

```diff
@@ -0,0 +1,223 @@
+    suffix = {
+      x86_64-linux = "linux";
+      aarch64-linux = "linux-arm64";
+      x86_64-darwin = "mac";
+      aarch64-darwin = "mac-arm64";
+    }.${system};
+    filename = "playwright-${driverVersion}-${suffix}.zip";
+  in stdenv.mkDerivation {
+    pname = "playwright-driver";
+    version = driverVersion;
+
+    src = fetchurl {
+      url = "https://playwright.azureedge.net/builds/driver/${filename}";
+      sha256 = {
+        x86_64-linux = "1nah45q9mihzkz2hqx6dwn7ac88vc22m5zspr23hc7dd1q4ll4ys";
+        aarch64-linux = "0wg4z2gb2fhvv4c2gjm6ncvbg9bkaavzq76x1bmm4lsnhjz312hp";
+        x86_64-darwin = "1nd6ll6dj0aqic3hh65y0br051ckbcap8spqbvzm19g8dn7vgv7m";
+        aarch64-darwin = "08qrzahmdhdv8flfjb58m27dqihq8jc4gwix6knc71209a9x8b83";
+      }.${system};
+    };
+    sourceRoot = ".";
+
+    nativeBuildInputs = [ unzip ];
+    buildInputs = [ nodejs ];
+
+    postPatch = ''
+      # Use Nix's NodeJS instead of the bundled one.
+      substituteInPlace playwright.sh --replace '"$SCRIPT_PATH/node"' '"${nodejs}/bin/node"'
+      rm node
+
+      # Hard-code the script path to be our $out directory. The alternative would be to add
+      # coreutils to the closure and use realpath like the upstream script does.
+      substituteInPlace playwright.sh --replace \
+        'SCRIPT_PATH="$(cd "$(dirname "$0")" ; pwd -P)"' \
+        "SCRIPT_PATH=$out"
+
+      patchShebangs playwright.sh
+      patchShebangs package/bin/*.sh
+    '';
+
+    dontBuild = true;
+
+    installPhase = ''
+      mkdir -p $out/bin
+      mv playwright.sh package $out/
+
+      # The actual driver script needs to stay at $out/playwright.sh because the Python bindings
+      # expect it to be there. But we can add a symlink so it's available in PATH as well.
+      ln -s $out/playwright.sh $out/bin/playwright-driver
+    '';
+
+    passthru = {
+      inherit filename;
+    };
+  };
+
+  browsers-mac = stdenv.mkDerivation {
+    pname = "playwright-browsers";
+    version = driverVersion;
+
+    src = runCommand "playwright-browsers-base" {
+      outputHashMode = "recursive";
+      outputHashAlgo = "sha256";
+      outputHash = {
+        x86_64-darwin = "1nd6ll6dj0aqic3hh65y0br051ckbcap8spqbvzm19g8dn7vgv7m";
+      }.${system};
+    } ''
+      export PLAYWRIGHT_BROWSERS_PATH=$out
+      ${driver}/playwright.sh install
+      rm -r $out/.links
+    '';
+
+    dontConfigure = true;
+    dontBuild = true;
+
+    installPhase = ''
+      mkdir $out
+      cp -r * $out/
+    '';
+
+    doCheck = false;
+  };
+
+  browsers-linux = let
+    fontconfig = makeFontsConf {
+      fontDirectories = [];
+    };
+  in runCommand "playwright-browsers" {
+    nativeBuildInputs = [ makeWrapper ];
+  } ''
+    mkdir -p $out/chromium-1005/chrome-linux $out/ffmpeg-1007
+
+    # See here for the Chrome options:
+    # https://github.com/NixOS/nixpkgs/issues/136207#issuecomment-908637738
+    makeWrapper ${chromium}/bin/chromium $out/chromium-1005/chrome-linux/chrome \
```

Comment:
This path is also very suboptimal and the version number will probably change in the future and then things will break.

Reply:
I'm very new to Nix, so I'm been playing with three different repos trying to get it all to work (in particular Firefox, Chromium works, thanks to the contributors here)

This file determines the version, would it be possible to 'grep' (parse) it in order to automatically extract the expected numbers?
It's simple to find,
`https://raw.githubusercontent.com/microsoft/playwright/v${playwright_version}/packages/playwright-core/browsers.json`
becomes
https://raw.githubusercontent.com/microsoft/playwright/v1.25.0/packages/playwright-core/browsers.json

And extracting Chromium (1019) or Firefox (1344) with even the associated/expected browser version should be easy enough for someone that knows what they are doing (?)

https://raw.githubusercontent.com/microsoft/playwright/v1.25.0/packages/playwright-core/browsers.json

In general, this repo is worth a view: https://github.com/voidus/nix-playwright-poc

Reply:
We could extract it with something like which would be way better.

``jq -r '.browsers[] | select(.name == "chromium").revision' packages/playwright-core/browsers.json``